core/src/main/java/org/apache/iceberg/view/ViewRepresentationParser.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+class ViewRepresentationParser {
+  static final String TYPE = "type";
+
+  private ViewRepresentationParser() {}
+
+  static void toJson(ViewRepresentation representation, JsonGenerator generator)
+      throws IOException {
+    Preconditions.checkArgument(representation != null, "Invalid view representation: null");
+    switch (representation.type()) {
+      case ViewRepresentation.Type.SQL:
+        SQLViewRepresentationParser.toJson((SQLViewRepresentation) representation, generator);
+        break;
+
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "Cannot serialize unknown view representation type %s", representation.type()));
+    }
+  }
+
+  static String toJson(ViewRepresentation entry) {
+    return JsonUtil.generate(gen -> toJson(entry, gen), false);
+  }
+
+  static ViewRepresentation fromJson(String json) {
+    return JsonUtil.parse(json, ViewRepresentationParser::fromJson);
+  }
+
+  static ViewRepresentation fromJson(JsonNode node) {
+    Preconditions.checkArgument(node != null, "Cannot parse view representation from null object");
+    Preconditions.checkArgument(
+        node.isObject(), "Cannot parse view representation from non-object: %s", node);
+    String type = JsonUtil.getString(TYPE, node).toLowerCase(Locale.ENGLISH);
+    switch (type) {
+      case "sql":
+        return SQLViewRepresentationParser.fromJson(node);
+
+      default:
+        throw new IllegalArgumentException(
+            String.format("Cannot deserialize unknown view representation type %s", type));
```

Comment:
Unknown representations should not cause a failure. Instead, I think that we want to have an `UnknownRepresentation` (similar to `UnknownTransform`). That way, we can still read a view that has a SQL representation and another representation and this is forward-compatible.

If there is an `UnknownRepresentation` then the view should be read-only because we don't want to drop or corrupt representations. (We could also keep the `JsonNode` around and reserialize it, but it's easier for now to make the view read-only.)

Reply:
This makes sense after reading how UnknownTransform works. I've updated the PR so that when reading, any unknown representation is deserialized into an UnknownViewRepresentation with a String type. When serializing the metadata we will still fail. This will effectively make the View read only if there's a single unknown representation and we can update later to serialize if desired